tachyon/zk/lookup/halo2/verifier.h

```diff
@@ -17,36 +17,44 @@
 #include "tachyon/zk/lookup/verifier.h"
 #include "tachyon/zk/lookup/verifying_evaluator.h"
 #include "tachyon/zk/plonk/base/l_values.h"
+#include "tachyon/zk/plonk/halo2/proof.h"
 
 namespace tachyon::zk::lookup {
 namespace halo2 {
 
 template <typename F, typename C>
 class Verifier final : public lookup::Verifier<typename halo2::Verifier<F, C>> {
  public:
-  explicit Verifier(const VerifierData<F, C>& data) : data_(data) {}
+  using Proof = plonk::halo2::Proof<F, C>;
 
-  void Evaluate(const std::vector<lookup::Argument<F>>& arguments,
-                const plonk::LValues<F>& l_values, std::vector<F>& evals) {
+  Verifier(const Proof& proof, size_t circuit_idx)
+      : data_(proof.ToLookupVerifierData(circuit_idx)) {}
+
+  Verifier(const Proof& proof, size_t circuit_idx,
+           const plonk::LValues<F>& l_values)
+      : data_(proof.ToLookupVerifierData(circuit_idx)), l_values_(&l_values) {}
```

Comment:
```suggestion
           const plonk::LValues<F>* l_values)
      : data_(proof.ToLookupVerifierData(circuit_idx)), l_values_(l_values) {}
```

Reply:
I made this constructor to take reference so that user can simply put `l_values` itself not a pointer in instantiation.
If I apply your suggestion the instantiation should be modified like this
```
// before
LookupVerifier lookup_verifier(proof, i, l_values);

// modified
LookupVerifier lookup_verifier(proof, i, &l_values);
```